src/solution_check.jl

```diff
@@ -0,0 +1,191 @@
+function check_range_bounds(ranges::Vector{Range}, target_value::Float64)
+    
+    for range in ranges
+        if target_value <= range.ub && target_value >= range.lb
+            return true
+        end
+    end
+    return false
+
+end
+
+function check_sequence(route::Route, data::RvrpInstance,
+                        computed_data::RvrpComputedData, solution::RvrpSolution,
+                        complete_req_ids::Set{String},
+                        ongoing_req_ids::Set{String})
+
+    travel_times = data.travel_specifications[1].travel_time_matrix
+    act_idx = 1
+    prev_act = route.sequence[1]
+    v_set = data.vehicle_sets[computed_data.vehicle_set_id_2_index[route.vehicle_set_id]]
+    v_category = data.vehicle_categories[computed_data.vehicle_category_id_2_index[v_set.vehicle_category_id]]
+    vehicle_properties = v_category.vehicle_properties
+    vehicle_capacities = v_category.vehicle_capacities
+    used_capacity = Dict{String,Float64}(
+        k => 0.0 for k in keys(vehicle_capacities)
+    )
+
+    for act_idx in 1:length(route.sequence)
+        act = route.sequence[act_idx]
+        if (act.operation_type != 0 && act.request_id == ""
+            || act.operation_type == 0 && act.request_id != "")
+            error("In action ", act.id, " of route ", route.id,
+                  ": operation type (", act.operation_type, ") and reques id (",
+                  act.request_id, " are not consistent.")
+        end
+
+        if act.operation_type in [1, 2] # Pickup or delivery (has req.id)
+            req = data.requests[computed_data.request_id_2_index[act.request_id]]
+            product_specification = data.product_specification_classes[computed_data.product_specification_class_id_2_index[req.product_specification_class_id]]
+            # Check properties
+            for (k,v) in product_specification.property_requirements
+                if v < vehicle_properties[k]
+                    error("In action ", act.id, " of route ", route.id,
+                          ": Uses property (", k, ") that vehicle ",
+                          v_set.id, " does not have. Vehicle properties: ",
+                          vehicle_properties, ".")
+                end
+            end
+            if act.operation_type == 1 || act.operation_type == 2
```

Comment:
This condition is not needed as it is already tested in line 37 https://github.com/atoptima/RichVehicleRoutingProblem.jl/pull/32/files#diff-bca8ce6d51b53a49a7c6ff082335b8a7R37